hive-metastore/src/main/java/org/apache/iceberg/hive/HiveClientPool.java

```diff
@@ -83,6 +83,15 @@ protected HiveMetaStoreClient reconnect(HiveMetaStoreClient client) {
     return client;
   }
 
+  @Override
+  protected boolean failureDetection(Exception e) {
+    if (super.failureDetection(e) || (e != null && e instanceof MetaException &&
+            e.getMessage().contains("Got exception: org.apache.thrift.transport.TTransportException"))) {
```

Comment:
Nit: continuation indents are 2 indents / 4 spaces.

Reply:
It's solved.